src/main/java/com/bnorm/auto/dispatch/internal/Writer.java

```diff
@@ -108,9 +112,38 @@ public static JavaFile write(MethodClassDescriptor methodClassDescriptor) {
         codeBlockBuilder.addStatement("throw new NullPointerException($S)", "dispatch value == null");
         first = false;
       }
-      for (MultiDescriptor descriptor : methodDescriptor.multiMethods()) {
+
+      // Work with multiMethod dispatch values as a list
+      // This allows a multiMethod to accept several dispatch values
+      Function<MultiDescriptor, List<?>> toDispatchValues = descriptor ->
+          descriptor.value().getValue() instanceof List<?> ? (List<?>) descriptor.value().getValue()
```

Comment:
Also, I want to avoid these levels of indirection.  Eventually, I would like to the descriptors as containing all the information needed directly.  So I would add the value to the descriptor as a property instead of indirectly retrieving it.  Make the discover phase build the complete object needed for the write phase.